src/index.ts

```diff
@@ -50,29 +49,38 @@ export interface ChangeableTrackerOptions {
   serviceProps?: ServiceProp;
 }
 
-interface PageViewQueueItem {
-  type: "pageview";
-  ts: Date;
-  href: string;
-  referrer?: string;
-}
+type EventParameter = Parameters<EventTracker[keyof EventTracker]>;
+type QueueItem = { consumerMethod: keyof EventTracker } & EventParameter;
 
-interface EventQueueItem {
-  type: "event";
-  ts: Date;
-  name: string;
-  data: any;
-}
 
-type QueueItem = PageViewQueueItem | EventQueueItem | { type: "impression" | "registration" };
+function pushEventToQueue(consumerMethodName?: keyof EventTracker) {
+  return (target: any, propertyKey: keyof EventTracker, descriptor: PropertyDescriptor) => {
+
+    consumerMethodName = consumerMethodName || propertyKey
+
+    const originalMethod = descriptor.value;
+
+    descriptor.value = function () {
+      const context = this
+      const eventRecord: QueueItem = originalMethod.apply(context, arguments);
+
+      eventRecord.consumerMethod = consumerMethodName;
```

Comment:
`item` 이 `EventParameter` 이면서 거기에 `consumerMethod` 속성도 가질 수 있도록 한 것 같은데, 그러다 보니 나중에 `runTrackersMethod` 호출하기 전에 `consumerMethod` 를 삭제해야 하는 등 부자연스러운 것 같습니다.

다음과 같이 `QueueItem` 이 `EventParameter` 와 `consumerMethod` 를 둘 다 속성으로 갖게 하는게 낫지 않을까요?
```typescript
type QueueItem = {
  consumerMethod: keyof EventTracker;
  eventParams: EventParameter;
}
```

그러면 여기서는 다음과 같이 바꿀 수 있을 것이고,
```suggestion
      const eventParams: EventParamter = originalMethod.apply(context, arguments);
      const eventRecord: QueueItem = new QueueItem(consumerMethodName, eventParams);
```

`runTrackerMethod` 에서는 다음과 같이 바꿀 수 있겠죠.
```typescript
  private runTrackersMethod(item: QueueItem): void {
    this.initializedTrackers().forEach(t => {
      const trackerMethod = (t as any)[item.consumerMethod];
      const args = Object.values(item.eventParams);
      trackerMethod.apply(t, args);```

Reply:
오.. 이렇게 수정하니 훨씬 자연스러운 것 같네요!!